src/canvas/Renderer.tsx

```diff
@@ -241,7 +243,7 @@ export class Renderer implements IRenderer {
             this._currPath.snapEndpoints(this._document.getPatternPaths());
 
             if (callback) {
-                callback();
+                this._currPath.addPoint(callback());
```

Comment:
I don't think that this is a good idea. The idea of the call back is that it is flexible and you could pass it any function. Now you are requiring the whatever the callback is, it has to return a point that should be added.